nemo/collections/nlp/modules/common/megatron/utils.py

```diff
@@ -442,8 +442,11 @@ def get_iterator_k_split(batch: Union[Dict, List[torch.Tensor]], num_microbatche
 
         # Split tensor items
         items = list(tensor_items.items())
-        assert items[0][1].shape[0] % num_microbatches == 0, "Issue with batch size configuration!"
```

Comment:
need someone from NLP to review the change on removing the constraint `assert items[0][1].shape[0] % num_microbatches == 0`

Reply:
acknowledge

Reply:
is there a way we can separate the cases where we know the batch size is not divisible with cases where things should be divisible?

silently rounding off can cause silent errors which can cause accuracy regressions and be super difficult to figure out

Reply:
looks good, thanks!